### DISCUSSION TITLE
steer only wake free turbines

### DISCUSSION THREAD

**Discussion by aclerc** [NatLabRockies/floris#980] [Ideas]
I've written some code to check which turbines are wake free (based on criteria in IEC61400-12-1:2017) and only let those turbines steer. I do that by editing the `minimum_yaw_angle` and `maximum_yaw_angle` arrays passed to `YawOptimizationSR` and running the wake free check and optimizer one direction at a time. This is particularly useful for designing the look up table for a validation campaign where it's difficult to measure wake steering uplift deep in the wind farm. Just wondering if this is already a feature I've missed? Happy to contribute code related to this if it's useful.

**Comment by misi9170** [comment]
Hi @aclerc , thanks for asking! In my opinion, this type of "practical consideration" for wake steering is probably a little too niche for FLORIS. We've been trying recently to keep FLORIS more general and limit the number of special cases that go in, and my feeling is that this is probably enough of a special case that I'd probably leave it out (but others, feel free to disagree with me).

That being said, I think this could be a nice additional feature to [FLASC](https://github.com/NREL/flasc/). In FLASC, we have a [utility for handling practical modifications to a FLORIS-designed yaw lookup table](https://github.com/NREL/flasc/blob/main/flasc/utilities/lookup_table_tools.py), which your approach could possibly be added to as a new function.

Alternatively, I wonder whether an optimal lookup schedule from FLORIS (formatted as a Dataframe `df_opt`) for all turbines could simply be "clipped" to just the freestream turbines---that is, rather than designing a lookup table assuming that only freestream turbines steer, it may be possible to design one where all turbines steer, and then essentially mask it to only the freestream turbines (possibly using [`get_upstream_turbs_floris`](https://github.com/NREL/flasc/blob/main/flasc/utilities/floris_tools.py#L648)). This is a slightly different approach that could be suboptimal, but we've generally found that at least when using the serial refine method, the upstream turbines' yaw angles are insensitive to whether the downstream turbines are yaw misaligned or not. There is also some wasted computation here (determining yaw angles for downstream turbines that may not be finally used), but given the speed of the serial refine method, this may not be a big deal.
